### ISSUE TITLE
Roadmap of godot-python

### ISSUE THREAD

**Issue by batu** [touilleMan/godot-python#126]
Hello! 

I just came across Leblond's europython talk and it lead me to this repo. I am genuinely excited to see this because for a while now I had been looking for a way to bring in tensorflow/pytorch functionality to a fully fledged game engine. Before today I had been mostly struggling with Unity and failing to find an elegant way.

I know that there is a good lack of tooling that combines the new AI craze with *any* game engine and this might be very exciting to a lot of people.

On this note, I was wondering what the future plans for this project were, and maybe see if I could start contributing by making examples etc.

Thank you for this project so far!

**Comment by touilleMan** [comment]
Hi @batu !

Last time was not really active due to a very demanding daily job, things are getting better though so I'm back on the project now ;-)
In the meantime I've got my hands on Cython which seems like a really good option for this project (Cython allows you to compile a subset of python to endup with C-like performances).
On top of that there are some changes I'd like to make (mostly replace Godot API bindings generated at runtime by a more traditional static bindings which should lower code complexity and make the project compatible with the release version of godot).

So long story short, I'm currently working on [a new branch](https://github.com/touilleMan/godot-python/tree/cython-rewrite). This is some massive changes so I guess it will take some time before the next release ;-)

Anyway I would gladly accept any help (even if you don't have previous experience with Cython or Python bindings, we are here only to have fun after all ^^) you can catch me on discord to discuss this further (I'm touilleMan on the [GodotEngine](https://discordapp.com/channels/212250894228652034/342045559768481802) discord)

**Comment by fuderiki** [comment]
Hi Touille-man,

I was just coming to ask a little how the cython implementation is going ...
I've been able to make quite some cool stuff until now with the current python bindings in Godot, but I must say that the supplementary juice that a Cython implementation could bring gets me quite excited.

Not rushing you though, ;)

just checking how things are going.

Courage on your amazing task, what you are doing is gold!

**Comment by touilleMan** [comment]
Hi @fuderiki ;-)

The Cython rework is doing really good !

- the pythonscript module is loaded by Godot
- string and variant types can be converted between Godot and Python
- Vector2, Array and Dictionary builtins are implemented
- the [binding generator](https://github.com/touilleMan/godot-python/blob/cython-rewrite/tools/generate_bindings.py) is working

Those are key points that pave the way to the next point which is to expose a Python class to Godot (it should be mostly copy/paste from the previous implementation so I'm optimistic on this ^^)

After that implementing the remaining builtins focus on the release !

At the beginning of february will be held the GodotCon and FOSDEM in Bruxelles, I hope I'd have something nice to show by then ;-)

**Comment by santagada** [comment]
This all seems really nice, I also saw there are new releases on this repo than on the godot assetstore, any reason for that? If not maybe updating that version, or putting a note there pointing here (and updating the readme here to point to its own releases) would be very nice.

**Comment by touilleMan** [comment]
@santagada yeah the version on the assetstore is pretty old (and doesn't event work with Godot 3.2) so we should definitely update this asap ;-)

However the master branch doesn't work with macOS right now so I would prefer to have it before doing the release...

But if people have a great need for this, we could put macOs on the side ^^

**Comment by apockill** [comment]
I would like to +1 a "great need" out of personal selfishness! It was so easy to use the version from the assetstore, and I'm having a difficult time debugging some issues with the build system to get the latest version.

**Comment by santagada** [comment]
Yes I really think having one of the main platform people use is better than nothing. Maybe even enticing for osx  people to help. If the powerbrick on my mac hadn't blew over a day before the start of quarantine I could even take a look myself.

**Comment by touilleMan** [comment]
@apockill @santagada I've released version 0.30.0, it should be soon on the assetstore (I've submitted it, but it must be reviewed)
Otherwise you can download per-platform builds here https://github.com/touilleMan/godot-python/releases/tag/v0.30.0
